### PR TITLE
added sys.allocations table to troubleshooting guide

### PR DESCRIPTION
**Pull request by chaudum**
MERGE ONLY WHEN FEATURE IS RELEASED

### CODE REVIEW

File: docs/best_practices/systables.txt [hunk 1]
```diff
@@ -225,7 +225,45 @@ list the hosts and tables with the highest number of rows inside a single
...
+===================================
+
+Related to the previous step about gaining insights about shards and
+replication is the step about cluster wide shard allocations.
+
+In some circumstances shards allocations behave different than you might expect
+to behave. A typical example is that a table remains underreplicated for some
+reason. In this case you may want to know more about the decision of the
+cluster why certain shards could not be allocated. For that, there is the
+``sys.allocations`` table. That table lists all shards in the cluster.
```
Comment (nslater) [inline docs/best_practices/systables.txt]: "table, which lists all shards in the cluster."

File: docs/best_practices/systables.txt [hunk 2]
```diff
@@ -225,7 +225,45 @@ list the hosts and tables with the highest number of rows inside a single
...
+
+If a shard is unassigned, it includes a reason why it cannot be allocated
```
Comment (nslater) [inline docs/best_practices/systables.txt]: "the row will also..."

File: docs/best_practices/systables.txt [hunk 3]
```diff
@@ -225,7 +225,45 @@ list the hosts and tables with the highest number of rows inside a single
...
+including the allocation decisions of each node.
```
Comment (nslater) [inline docs/best_practices/systables.txt]: this bit is unclear. can you reword?

File: docs/best_practices/systables.txt [hunk 4]
```diff
@@ -225,7 +225,45 @@ list the hosts and tables with the highest number of rows inside a single
...
+
+If a shard is assigned but cannot be moved/rebalanced, it includes a reason why
```
Comment (nslater) [inline docs/best_practices/systables.txt]: "moved or rebalanced" if they're distinct operations, or pick one word

Comment (nslater) [reply docs/best_practices/systables.txt]: "it include" -> "the row includes"

File: docs/best_practices/systables.txt [hunk 5]
```diff
@@ -225,7 +225,45 @@ list the hosts and tables with the highest number of rows inside a single
...
+it remains on the current node.
+
+For a full list of available columns see the `reference documentation about
```
Comment (nslater) [inline docs/best_practices/systables.txt]: please move the link url to the bottom of the section

File: docs/best_practices/systables.txt [hunk 6]
```diff
@@ -225,7 +225,45 @@ list the hosts and tables with the highest number of rows inside a single
...
+system tables </docs/crate/reference/sql/system.html#Allocations>`__.
+
+To find out about the states and resons of shards of a specific table you can
```
Comment (nslater) [inline docs/best_practices/systables.txt]: reword "reasons of". not clear what is meant

**Comment by chaudum** [comment]
@nslater wanna have another look?

File: docs/best-practices/systables.txt [hunk 7]
```diff
@@ -225,7 +225,45 @@ list the hosts and tables with the highest number of rows inside a single
...
+Step 5: Analyze Allocation Problems
+===================================
+
+Related to the previous step about gaining insights about shards and
+replication is the step about cluster wide shard allocations.
+
+In some circumstances shards allocations behave different than you might expect
+to behave. A typical example is that a table remains underreplicated for some
+reason. In this case you may want to know more about the decision of the
+cluster why certain shards could not be allocated. For that, there is the
```
Comment (nslater) [inline docs/best-practices/systables.txt]: "In this case you may want to know more about the decision of the cluster why certain shards could not be allocated"

this is confusingly worded. can you rephrase?

File: docs/best-practices/systables.txt [hunk 8]
```diff
@@ -225,7 +225,45 @@ list the hosts and tables with the highest number of rows inside a single
...
+``sys.allocations`` table, which lists all shards in the cluster.
+
+If a shard is unassigned, the row will also include a reason why it cannot be
+allocated on any node.
+
+If a shard is assigned but cannot be moved or rebalanced, the row includes a
+reason why it remains on the current node.
+
+For a full list of available columns see the `reference documentation about
```
Comment (nslater) [inline docs/best-practices/systables.txt]: "columns, see"

File: docs/best-practices/systables.txt [hunk 9]
```diff
@@ -225,7 +225,45 @@ list the hosts and tables with the highest number of rows inside a single
...
+system tables`_.
+
+To find out about the different states of shards of a specific table you can
```
Comment (nslater) [inline docs/best-practices/systables.txt]: "table, you"

File: docs/best-practices/systables.txt [hunk 10]
```diff
@@ -255,3 +293,6 @@ to restore. Suppose you make nightly backups, the command::
 
 shows you last week's snapshots along with its name, the stored indices, and
 how long they took.
+
+
```
Comment (nslater) [inline docs/best-practices/systables.txt]: drop double empty new line

File: docs/best-practices/systables.txt [hunk 11]
```diff
@@ -225,7 +225,45 @@ list the hosts and tables with the highest number of rows inside a single
...
+Step 5: Analyze Allocation Problems
+===================================
+
+Related to the previous step about gaining insights about shards and
+replication is the step about cluster wide shard allocations.
+
+In some circumstances shards allocations behave different than you might expect
+to behave. A typical example is that a table remains underreplicated for some
+not obvious reason. Then you want to know the decisions of the cluster not to
```
Comment (nslater) [inline docs/best-practices/systables.txt]: "Then you want to know the decisions of the cluster not to allocate the shards"

this still doesn't parse well for me. can you rephrase? do you mean "you may want to know why the cluster has not allocated the shards" or "you may want to find out what is causing the cluster not to.."?

Comment (chaudum) [reply docs/best-practices/systables.txt]: you want to know the decisions the cluster made. isn't that the same as "why the cluster did not allocate shards" ?
do you have a suggestion for the wording?

File: docs/best-practices/systables.txt [hunk 12]
```diff
@@ -225,7 +225,45 @@ list the hosts and tables with the highest number of rows inside a single
...
+
+Step 5: Analyze Allocation Problems
+===================================
+
+Related to the previous step about gaining insights about shards and
+replication is the step about cluster wide shard allocations.
+
+In some circumstances shards allocations behave different than you might expect
+to behave. A typical example is that a table remains underreplicated for some
+not obvious reason. Then you want to find out what is causing the cluster not
```
Comment (nslater) [inline docs/best-practices/systables.txt]: "typical example is that" -> "typical example might be that"

"Then you want" -> "You would probably want"

"the cluster not to" -> "the cluster to not"

File: docs/best-practices/systables.txt [hunk 13]
```diff
@@ -225,7 +225,45 @@ list the hosts and tables with the highest number of rows inside a single
...
+
+Step 5: Analyze Allocation Problems
+===================================
+
+Related to the previous step about gaining insights about shards and
+replication is the step about cluster wide shard allocations.
+
+In some circumstances shards allocations behave different than you might expect
+to behave. A typical example might be that a table remains underreplicated for
+some not obvious reason. You would probably want to find out what is causing
```
Comment (nslater) [inline docs/best-practices/systables.txt]: "for no apparent reason"

File: docs/best-practices/systables.txt [hunk 14]
```diff
@@ -225,7 +225,45 @@ list the hosts and tables with the highest number of rows inside a single
...
 
-Step 5: Managing Snapshots
+
+Step 5: Analyze Allocation Problems
+===================================
+
+Related to the previous step about gaining insights about shards and
+replication is the step about cluster wide shard allocations.
+
+In some circumstances shards allocations behave different than you might expect
```
Comment (nslater) [inline docs/best-practices/systables.txt]: let's reword the first sentence:

"In some circumstances, shard allocations might behave differently than you expect."
